src/ops/custom.rs

```diff
@@ -24,6 +24,14 @@ pub struct OpaqueOp {
 }
 
 impl OpaqueOp {
+    /// Initialize a new named OpaqueOp
+    pub fn new(id: impl Into<SmolStr>, custom: Box<dyn CustomOp>) -> Self {
+        Self {
+            id: id.into(),
+            custom,
+        }
```

Comment:
Make this generic, and avoid exposing the internals.
```suggestion
    pub fn new(id: impl Into<SmolStr>, custom: impl CustomOp) -> Self {
        Self {
            id: id.into(),
            custom: Box::new(custom),
        }
```